File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void deleteNode(struct ListNode* node) {
         struct ListNode* NextNode = node->next;
         node->val = NextNode->val;
         node->next = NextNode->next;
         free(NextNode);
}

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
          ListNode Next_Node = node.next;
          node.val = Next_Node.val;
          node.next = Next_Node.next;
        //   Next_Node.next = null;
    }
}